shell/platform/windows/platform_handler.h

```diff
@@ -33,6 +34,10 @@ class PlatformHandler {
 
   virtual ~PlatformHandler();
 
+  // Send a request to the framework to test if a cancelable exit request
+  // should be canceled or honored.
+  virtual void RequestAppExit(const std::string& exit_type, int64_t exit_code);
```

Comment:
Is the exit type an enum on the Dart side? If so, we should probably decode the incoming lifecycle message to an enum and use that in these methods?